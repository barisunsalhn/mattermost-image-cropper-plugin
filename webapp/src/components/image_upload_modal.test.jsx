import React from 'react';

import {shallow} from 'enzyme';

import {Modal} from 'react-bootstrap';

import ImageUploadModal from './image_upload_modal';

describe('components/imageUploadModal', () => {
    const baseProps = {
        imgURL: 'test',
        show: true,
        aspectRatio: 16 / 9,
        closeModal: jest.fn(),
    };

    test('should match snapshot, modal not shown', () => {
        const show = false;
        const props = {...baseProps, show};
        const wrapper = shallow(<ImageUploadModal {...props}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<Modal/>)).toBe(false);
    });
    test('should match snapshot, modal shown', () => {
        const show = true;
        const props = {...baseProps, show};
        const wrapper = shallow(<ImageUploadModal {...props}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(Modal).prop('show')).toBe(true);
    });
    test('should match snapshot after clicking upload full button, modal not shown', () => {
        const show = true;
        const props = {...baseProps, show};
        const wrapper = shallow(<ImageUploadModal {...props}/>);
        wrapper.find('#uploadFullButton').simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<Modal/>)).toBe(false);
    });
    test('should match snapshot after clicking crop button, modal not shown', () => {
        const show = true;
        const props = {...baseProps, show};
        const wrapper = shallow(<ImageUploadModal {...props}/>);
        wrapper.find('#cropButton').simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<Modal/>)).toBe(false);
    });
},

);