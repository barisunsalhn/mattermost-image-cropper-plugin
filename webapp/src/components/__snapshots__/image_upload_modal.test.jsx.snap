// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/imageUploadModal should match snapshot after clicking crop button, modal not shown 1`] = `
<Modal
  animation={true}
  aria-labelledby="imageUploadModalLabel"
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  dialogClassName="modal-image image"
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    className="image"
    closeButton={true}
    closeLabel="Close"
  >
    <div>
      Please crop your image or screenshot pasted...
    </div>
  </ModalHeader>
  <ModalBody
    className="image"
  >
    <div>
      <Cropper
        aspect={1.7777777777777777}
        classes={
          Object {
            "containerClassName": "container",
          }
        }
        crop={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cropShape="rect"
        image="test"
        maxZoom={3}
        mediaProps={Object {}}
        minZoom={1}
        objectFit="contain"
        onCropChange={[Function]}
        onCropComplete={[Function]}
        onZoomChange={[Function]}
        restrictPosition={true}
        rotation={0}
        showGrid={false}
        style={Object {}}
        zoom={1}
        zoomSpeed={1}
        zoomWithScroll={true}
      />
    </div>
  </ModalBody>
  <ModalFooter>
    <button
      className="btn btn-primary save-button"
      id="uploadFullButton"
      onClick={[Function]}
      type="button"
    >
      Upload full
    </button>
    <button
      className="btn btn-primary save-button"
      id="cropButton"
      onClick={[Function]}
      type="button"
    >
      Crop
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/imageUploadModal should match snapshot after clicking upload full button, modal not shown 1`] = `
<Modal
  animation={true}
  aria-labelledby="imageUploadModalLabel"
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  dialogClassName="modal-image image"
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    className="image"
    closeButton={true}
    closeLabel="Close"
  >
    <div>
      Please crop your image or screenshot pasted...
    </div>
  </ModalHeader>
  <ModalBody
    className="image"
  >
    <div>
      <Cropper
        aspect={1.7777777777777777}
        classes={
          Object {
            "containerClassName": "container",
          }
        }
        crop={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cropShape="rect"
        image="test"
        maxZoom={3}
        mediaProps={Object {}}
        minZoom={1}
        objectFit="contain"
        onCropChange={[Function]}
        onCropComplete={[Function]}
        onZoomChange={[Function]}
        restrictPosition={true}
        rotation={0}
        showGrid={false}
        style={Object {}}
        zoom={1}
        zoomSpeed={1}
        zoomWithScroll={true}
      />
    </div>
  </ModalBody>
  <ModalFooter>
    <button
      className="btn btn-primary save-button"
      id="uploadFullButton"
      onClick={[Function]}
      type="button"
    >
      Upload full
    </button>
    <button
      className="btn btn-primary save-button"
      id="cropButton"
      onClick={[Function]}
      type="button"
    >
      Crop
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/imageUploadModal should match snapshot, modal not shown 1`] = `""`;

exports[`components/imageUploadModal should match snapshot, modal shown 1`] = `
<Modal
  animation={true}
  aria-labelledby="imageUploadModalLabel"
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  dialogClassName="modal-image image"
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    className="image"
    closeButton={true}
    closeLabel="Close"
  >
    <div>
      Please crop your image or screenshot pasted...
    </div>
  </ModalHeader>
  <ModalBody
    className="image"
  >
    <div>
      <Cropper
        aspect={1.7777777777777777}
        classes={
          Object {
            "containerClassName": "container",
          }
        }
        crop={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cropShape="rect"
        image="test"
        maxZoom={3}
        mediaProps={Object {}}
        minZoom={1}
        objectFit="contain"
        onCropChange={[Function]}
        onCropComplete={[Function]}
        onZoomChange={[Function]}
        restrictPosition={true}
        rotation={0}
        showGrid={false}
        style={Object {}}
        zoom={1}
        zoomSpeed={1}
        zoomWithScroll={true}
      />
    </div>
  </ModalBody>
  <ModalFooter>
    <button
      className="btn btn-primary save-button"
      id="uploadFullButton"
      onClick={[Function]}
      type="button"
    >
      Upload full
    </button>
    <button
      className="btn btn-primary save-button"
      id="cropButton"
      onClick={[Function]}
      type="button"
    >
      Crop
    </button>
  </ModalFooter>
</Modal>
`;
